### PR TITLE
Implement some more comments

### PR DESCRIPTION
**Pull request by tharvik**
* fix scroll on Chrome (and probably Safari)
* use ₳ as symbol
* open links in new tab
* improve text & some more small changes

### CODE REVIEW

File: web/src/pages/speedtest.rs [hunk 1]
```diff
@@ -245,60 +207,62 @@ impl Component for Speedtest {
 
             <p> { "
                 One of the most interesting features of AT2 is that it allows
-                for many more transactions per second (TPS) than the most
+                for many more transactions per second (TPS) than the two most
                 popular blockchains, such as Bitcoin or Ethereum.
-                Here you can actually test it, flooding the servers with
+                Indeed, it scales in " }
```
Comment (ineiti) [inline web/src/pages/speedtest.rs]: This part is not clear to me. On one hand it scales in `log(#nodes)`, but then you write that the TPS gets reduced with more nodes... So which one is it?

If I'm correct and it actually scales in `log(#nodes)`, I'd write something like:

```
The number of transactions per second that the system can validate is proportional to `log(#nodes)`. So the more nodes you add, the more transactions per second the system can validate. However, as you add more and more nodes, the increase in speed diminishes.
```

File: web/src/pages/speedtest.rs [hunk 2]
```diff
@@ -245,60 +207,62 @@ impl Component for Speedtest {
...
+                <code class="math">{ "log(#nodes)" }</code>
+                { " meaning that when you exponentially increase the number of
+                nodes, it only linearly reduce the TPS. The two others chains
+                scale in " }
+                <code class="math">{ "#nodes" }</code>
+                { " meaning that doubling the nodes divide the TPS by two.
```
Comment (ineiti) [inline web/src/pages/speedtest.rs]: Same as above - is that true?

File: web/src/pages/speedtest.rs (hunk 1 above)
Comment (ineiti) [inline web/src/pages/speedtest.rs]: And alos: third person singular most often takes an `s`: `linearly reduceS`, `doubling the nodes divideS`.

File: web/src/pages/summary.rs [hunk 3]
```diff
@@ -20,57 +20,59 @@ impl Component for Summary {
...
-                <a href="https://bitcoin.org"> { "Bitcoin" } </a>
+                It can send asset, show other accounts on the network
+                and watch the latest processed transactions.
+                But using a speedtest reveals the true power of AT2:
+                it is faster than other two most popular blockchains, beating " }
+                <a href="https://bitcoin.org" target="_blank"> { "Bitcoin" } </a>
                 { " and " }
-                <a href="https://ethereum.org"> { "Ethereum" } </a>
+                <a href="https://ethereum.org" target="_blank"> { "Ethereum" } </a>
                 { " by a factor of 10.
```
Comment (ineiti) [inline web/src/pages/summary.rs:34]: That's really not a lot. I thought that we saw a factor of 100, no? And theoretical performance is even more?

Also, latest developments in blockchains have more than 600kTPS: https://arxiv.org/abs/2105.11827

File: web/src/pages/summary.rs [hunk 4]
```diff
@@ -20,57 +20,59 @@ impl Component for Summary {
...
+                It can also be used as a means of communication, an
                 instant chat ensuring that no rogue participant can interfere
                 or reorder the messages.
             " } </p>
 
             <p> { "
-                Most blockchains are still using the global consensus method,
-                but AT2's paper demonstrates that this is unnecessary to
-                transfer asset within a network.
+                Most blockchains still uses the global consensus method,
```
Comment (ineiti) [inline web/src/pages/summary.rs]: now it's third person plural, so there is no `s`... `blockchains still use`.

File: web/src/pages/summary.rs [hunk 5]
```diff
@@ -20,57 +20,59 @@ impl Component for Summary {
...
+                but " }
+                <a href="https://arxiv.org/abs/1812.10844" target="_blank">
+                    { "AT2's research paper" }
+                </a> { " demonstrates that this is unnecessary to transfer asset
+                within a network, one only need local consensus to do so.
```
Comment (ineiti) [inline web/src/pages/summary.rs]: third person singular, takes an s: `only needS`

File: web/src/pages/summary.rs [hunk 6]
```diff
@@ -20,57 +20,59 @@ impl Component for Summary {
...
                 If you are interested in exploring this technology more, look at " }
-                <a href="https://factory.c4dt.org/showcase/at2">
+                <a href="https://factory.c4dt.org/showcase/at2" target="_blank">
                     { "the project page" }
                 </a> { "
                 and contact the "
-                } <a href="mailto:factory@c4dt.org"> { "C4DT" } </a> { ",
-                we will be pleased to discuss it with you in greater details."
+                } <a href="mailto:factory@c4dt.org"> { "C4DT" } </a> { ".
+                We will be pleased to discuss it with you further."
```
Comment (ineiti) [inline web/src/pages/summary.rs]:
```suggestion
                We're looking forward to have your input."
```

File: web/src/pages/your_account.rs [hunk 7]
```diff
@@ -0,0 +1,299 @@
...
+                            sequence,
+                            user_to_send_to,
+                            amount,
+                        ));
+
+                        self.props.bump_sequence.emit(sequence);
+                    }
+                }
+
+                // TODO in background
```
Comment (ineiti) [inline web/src/pages/your_account.rs]: is this an important todo?

File: web/src/pages/your_account.rs [hunk 8]
```diff
@@ -0,0 +1,299 @@
...
+                self.get_balance_agent
+                    .send(self.props.user.0.clone().to_thin());
+
+                false
+            }
+            Message::TransactionSent(ret) => {
+                ret.unwrap(); // TODO send asset in dialog
```
Comment (ineiti) [inline web/src/pages/your_account.rs]: important todo? Is it OK to `unwrap` here?

File: web/src/pages/your_account.rs [hunk 9]
```diff
@@ -0,0 +1,299 @@
...
+    }
+
+    fn view(&self) -> Html {
+        let now = Utc::now();
+
+        html! { <>
+            <h1> { "Your account" } </h1>
+
+            <p> { "
+                Now, your account is registered on the chain.
```
Comment (ineiti) [inline web/src/pages/your_account.rs]:
```suggestion
                Now your account is registered on the chain.
```

File: web/src/pages/your_account.rs [hunk 10]
```diff
@@ -0,0 +1,299 @@
...
+                Same as Bitcoin, you have a wallet, to which we added 100'000
```
Comment (ineiti) [inline web/src/pages/your_account.rs]:
```suggestion
                Similar to Bitcoin, you have a wallet, to which we added 100'000
```

File: web/src/pages/your_account.rs [hunk 11]
```diff
@@ -0,0 +1,299 @@
...
+                assets.
```
Comment (ineiti) [inline web/src/pages/your_account.rs]: Still need a better name for this.

File: web/src/pages/your_account.rs [hunk 12]
```diff
@@ -0,0 +1,299 @@
...
+            " } <br /> { "
+                Below, you can play by sending some asset to the other accounts
```
Comment (ineiti) [inline web/src/pages/your_account.rs]:
```suggestion
                Below, you can send some assets to other accounts
```

File: web/src/pages/your_account.rs [hunk 13]
```diff
@@ -0,0 +1,299 @@
...
+                of the network.
+                You can also see the transactions currently happening on the network.
+            " } <br /> { "
+                If you don't see your transactions, or that one fails, please " }
```
Comment (ineiti) [inline web/src/pages/your_account.rs]:
```suggestion
                If you don't see your transactions, or one fails, please " }
```

File: web/src/pages/your_account.rs [hunk 14]
```diff
@@ -0,0 +1,299 @@
...
+                <a href="mailto:factory@c4dt.org"> { "contact the C4DT" } </a>
+                { "." }
+            </p>
+
+            <hr />
+
+            <h2> { "Send assets" } </h2>
+
+            <p>{ "
+                Here, you can send some amount of asset to another account.
```
Comment (ineiti) [inline web/src/pages/your_account.rs]:
```suggestion
                Here you can send some amount of assets to another account.
```

**Review by ineiti — APPROVED** [review]
I didn't go into the programming part - I would need some more comments at least at the beginning of the files or the structures, to describe where and how they're used.

File: web/src/pages/speedtest.rs (hunk 1 above)
Comment (tharvik) [reply web/src/pages/speedtest.rs]: It's not the case, more nodes means less TPS, as there is more nodes to sample.
 * trivial case, a single server instantly validate a transaction
 * the more there is, the bigger the set of peers to sample, so more messages to send/recv

A more precise definition would be: the number of messages scales in `log(#nodes)`. But that's not smth I wrote about, the only metric used is the TPS.

I understand "scale"  as simply "how the system behave when adding more [smth]", not that it is faster necessarely.

Comment (ineiti) [reply web/src/pages/speedtest.rs]: So what about

```
The number of transactions per second that the system can validate is proportional to `1/log(#nodes)`. So the number of transactions per second the system can validate goes down as more nodes join the system. However, as you add more and more nodes, the change in speed gets less and less important.
```

File: web/src/pages/your_account.rs (hunk 11 above)
Comment (ineiti) [reply web/src/pages/your_account.rs]: So it seems we're using `assets` now. Which is fine.

File: web/src/pages/speedtest.rs (hunk 2 above)
Comment (tharvik) [reply web/src/pages/speedtest.rs]: hum, no, that's wrong. TPS is actually fixed in Bitcoin (and Ethereum IMO), having a block (with a max size) every 10min. So Bitcoin doesn't scale at all.

Yeah, so I added that after discussing with Olivier, but it falls short (he really wanted the "scales in log(n)"). I can go about it explaining that at2 only need to sample the network and not validate txs with each node, would that be clearer?

File: web/src/pages/summary.rs (hunk 3 above)
Comment (tharvik) [reply web/src/pages/summary.rs:34]: Yep, I know, I wrote a paragraph in the speedtest part to say that it's not best condition to test (via browser to node, ...).

The only measure I could find in the paper is
> We compared the performance of AT2 to that of a transfer system based on BFT-Smart, a state-of-the-art consensus-based state machine replication system. AT2 provides performance superior to that of the consensus-based solution. In systems of up to 100 replicas, regardless of system size, we observed a throughput improvement ranging from 1.5x to 6x, while achieving a decrease in latency of up to 2x.

File: web/src/pages/your_account.rs (hunk 7 above)
Comment (tharvik) [reply web/src/pages/your_account.rs]: Nope, dangling when I used a dialog.

File: web/src/pages/your_account.rs (hunk 8 above)
Comment (tharvik) [reply web/src/pages/your_account.rs]: Dangling from dialog, but yeah, let's not `unwrap`, the sent tx appears in the txs list anyway,

File: web/src/pages/your_account.rs (hunk 11 above)
Comment (tharvik) [reply web/src/pages/your_account.rs]: The icon directly?

File: web/src/pages/speedtest.rs (hunk 1 above)
Comment (tharvik) [reply web/src/pages/speedtest.rs]: I added:
> Indeed, its TPS scales in `1/log(#nodes)` meaning that the TPS goes down as more nodes join the network but the change in speed gets less and less important.

File: web/src/pages/speedtest.rs (hunk 2 above)
Comment (ineiti) [reply web/src/pages/speedtest.rs]: If he wants to `log(n)`, we can put it somewhere :) But `scales in log(n)` definitely means that the higher the n, the higher the throughput.

So it's `scales in 1/log(n)`, if you want so.

I wouldn't go into too much detail here. But this is also something you can discuss with Mattheo if you show him the demonstrator.

File: web/src/pages/your_account.rs (hunk 11 above)
Comment (ineiti) [reply web/src/pages/your_account.rs]: Trying out things: `100'000:icon: assets`?

Comment (tharvik) [reply web/src/pages/your_account.rs]: Going for: 100'000 ₳ (assets)

File: web/src/pages/speedtest.rs (hunk 2 above)
Comment (tharvik) [reply web/src/pages/speedtest.rs]: Going for: scales in `log(n)`; and add that the other chains have a fixed TPS.

File: web/src/pages/summary.rs (hunk 3 above)
Comment (tharvik) [reply web/src/pages/summary.rs:34]: Not changing it then.

**Pull request merged**